### DISCUSSION TITLE
onConflictAction: 'ignore' shoul not include `on conflict` colums if there is no provided onConflictFields

### DISCUSSION THREAD

**Discussion by wenerme** [mikro-orm/mikro-orm#6007] [General]
### Describe the bug

for

```ts
await em.upsert(
  DemoEntity,
  { md5: 'md5', sha256: 'sha256' },
  {
    onConflictAction: 'ignore',
  },
);

```

I hope to get 

```sql
insert into "demo" ("md5", "sha256") values ('md5', 'sha256') on conflict do nothing returning "tid", "eid", "uid", "md5"
```

**instead of**

```sql
insert into "demo" ("md5", "sha256") values ('md5', 'sha256') on conflict ("id") do nothing returning "tid", "eid", "uid", "md5"
```

### Reproduction

```bash
git clone https://github.com/wenerme/mikro-orm-issues
cd mikro-orm-issues
pnpm i

# failed with conflict
make tsx-run-is-2
```

### What driver are you using?

None

### MikroORM version

latest

### Node.js version

20

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
You can just add `onConflictFields: []` into the options.

> git clone https://github.com/wenerme/mikro-orm-issues

Also please don't open more issues with this repro, I already told you I don't consider those things you do there as supported, if you think you see an actual problem, reproduce it with just the ORM, no magic around mixins (or bun) involved.

**Reply by wenerme** [reply]
But this test include nothing about bun & mixing, just single file test with some helpers.

I don't know onConflictFields: [] this is required now, this breaks the old code.

**Reply by B4nan** [reply]
wdym old code, i dont think this has changed, it was always like that

**Reply by wenerme** [reply]
Sorry, my bad, I don't know this before, I thought the default is `on confilic` without column.
